liquidcore/config/views.py

```diff
@@ -0,0 +1,239 @@
+import fcntl
+from contextlib import contextmanager
+from django.conf import settings
+from django.contrib.auth.models import User
+from rest_framework.response import Response
+from rest_framework.permissions import IsAuthenticated, AllowAny, IsAdminUser
+from rest_framework import viewsets
+from rest_framework.decorators import detail_route, list_route, api_view
+from rest_framework.views import APIView
+from rest_framework import status
+
+from .models import *
+from .serializers import *
+from .system import update_system
+
+class UserViewSet(viewsets.ModelViewSet):
+    queryset = User.objects.all()
+    lookup_field = 'username'
+    lookup_value_regex = USERNAME_URL_REGEX
+
+    # don't let the user update the `username` field
+    def get_serializer_class(self):
+        if self.request.method == 'POST':
+            return CreateUserSerializer
+        return UpdateUserSerializer
+
+    # @list_route creates an endpoint that doesn't contain the pk.
+    # We don't return a list, but that's not a problem.
+    @list_route(
+        methods=['get'],
+        permission_classes=[AllowAny],
+        url_name='whoami'
+    )
+    def whoami(sef, request):
+        """Gets the current logged in user.
+        Marks the result under the key 'is_authenticated'."""
+
+        username = request.user.username
+        if not username:
+            return Response({"is_authenticated": False})
+        queryset = User.objects.get(username=username)
+        data = UpdateUserSerializer(queryset, context={'request':request}).data
```

Comment:
Can't you just do `data = UpdateUserSerializer(request.user, context={'request': request}).data`?